libs/langchain-google-common/src/connection.ts

```diff
@@ -53,11 +53,28 @@ export abstract class GoogleConnection<
 
   async _clientInfoHeaders(): Promise<Record<string, string>> {
     const clientLibraryVersion = await this._clientLibraryVersion();
+    const clientInfo= await this._getclientInfo();
     return {
       "User-Agent": clientLibraryVersion,
+      "Client-Info": clientInfo.toString(),
     };
   }
 
+  async _getclientInfo(): Promise<Record<string, string>> {
```

Comment:
There is a lot of code duplication between this and `_clientLibraryVersion()`. Perhaps combine the two?